inst/stan/functions/obs_lpmf.stan

```diff
@@ -1,20 +1,32 @@
-real obs_lpmf(int[] dummy, int start, int end, int[,] obs, int[] sl,
+real obs_lpmf(int[] dummy, int start, int end, int[] obs, int[] sl, int[] csl,
               vector[] imp_obs, int[] sg, int[] st, int[,] rdlurd,
               vector srdlh, matrix ref_lh, int[] dpmfs, int ref_p, real phi) {
   real tar = 0;
   real tar_obs;
-  int t;
+  int init_cl = sl[start];
+  int start_n = csl[start] - sl[start];
+  int end_n = csl[end];
+  int n = end_n - start_n;
+  int snap_obs[n] = obs[(start_n + 1):end_n];
+  vector[n] exp_obs;
+  int start_snap, g, t, l;
+
   for (i in start:end) {
-    vector[sl[i]] exp_obs;
-    vector[sl[i]] rdlh;
+    g = sg[i];
+    t = st[i];
+    l = sl[i];
+    start_snap = csl[i] - start_n - l;
```

Comment:
Just an idea regarding readibility: if we are using this loop anyway, wouldn't it be more intuitive code to initialize `start_snap` at 0* and then increment it by `l` in every iteration?

*or 1, as you prefer - would need to change slice of `exp_obs` below as well

Reply:
And rather call it `start_index` or so, to make clear that this is referencing the index of the `exp_obs` vector?

Reply:
Good idea - will add this.